addon/components/discover-page/component.js

```diff
@@ -502,7 +502,7 @@ export default Ember.Component.extend(Analytics, hostAppName, {
             }
         });
         // For PREPRINTS and REGISTRIES. If theme.isProvider, add provider(s) to query body
-        if (this.get('themeProvider.name') !== null) {
+        if (this.get('themeProvider')) {
```

Comment:
It looks like this logic needs to be refactored. a bit otherwise https://github.com/CenterForOpenScience/ember-osf/pull/289/files#diff-ae3ef0b00a49fadec81c8b9b765a6d78R512 will never get hit, unless I'm missing something

Reply:
instead you can move `filters.push` into both `if` statements

Reply:
or I guess remove this line completely, and rewrite L519:
```
 if (sources.length) {
    filters.push({
        terms: {
            sources: sources
        }
    });
}
```

Reply:
welp this is what I get for not reading. `themeProvider` != `theme.isProvider`. Ignore my last few comments.